Modules/_io/_iomodule.c

```diff
@@ -368,6 +367,14 @@ _io_open_impl(PyObject *module, PyObject *file, const char *mode,
         if (blksize_obj == NULL)
             goto error;
         buffering = PyLong_AsLong(blksize_obj);
+        if (buffering > 8192 * 1024)
```

Comment:
```suggestion
        if (buffering > MAXIMUM_BUFFER_SIZE)
```
and put `MAXIMUM_BUFFER_SIZE` in the header file next to `DEFAULT_BUFFER_SIZE`?